common/src/main/java/com/microsoft/identity/common/internal/broker/AuthUxJavaScriptInterface.kt

```diff
@@ -0,0 +1,120 @@
+import com.microsoft.identity.common.internal.numberMatch.NumberMatchHelper
+import com.microsoft.identity.common.java.util.JsonUtil
+import com.microsoft.identity.common.logging.Logger
+
+/**
+ * JavaScript API to receive JSON string payloads from AuthUX in order to facilitate calling various
+ * broker methods.
+ */
+class AuthUxJavaScriptInterface {
+
+    // Store number matches in a static hash map
+    // No need to persist this storage beyond the current broker process, but we need to keep them
+    // long enough for AuthApp to call the broker api to fetch the number match
+    companion object {
+        val TAG = AuthUxJavaScriptInterface::class.java.simpleName
+        private const val JAVASCRIPT_INTERFACE_NAME = "ClientBrokerJS"
+
+        fun getInterfaceName() : String {
+            return JAVASCRIPT_INTERFACE_NAME
+        }
+    }
+
+    /**
+     * Method to receive a JSON string payload from AuthUX through JavaScript API.
+     * Schema for the Json Payload:
+     *         {
+     *             "correlationID": "SOME_CORRELATION_ID" ,
+     *             "action_name":"write_data",
+     *             "action_component":"broker",
+     *             "params":
+     *             {
+     *                 "function": "NUMBER_MATCH",
+     *                 "data":
+     *                 {
+     *                     "sessionID": "$mockSessionId",
+     *                     "numberMatch": "$mockNumberMatchValue"
+     *                 }
+     *             }
+     *         }
+     * TODO: This is currently the schema set for numberMatch, there may be some additions made for
+     *  the more generalized JSON Schema for future Server-side to broker communication through JS.
+     *
+     * https://microsoft-my.sharepoint-df.com/:w:/p/veenasoman/EY1AZIeT8X5KrXVz97Vx520B3Jj0fBLSPlklnoRvcmbh0Q?e=VzNFd1&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2Cfadidurah%40microsoft.com&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI0OS8yNTA1MDQwMTYwOSIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D
+     */
+    @JavascriptInterface
+    fun postMessageToBroker(jsonPayload: String) {
+        val methodTag = "$TAG:postMessageToBroker"
+        Logger.info(methodTag, "Received a payload from AuthUX through JavaScript API.")
+
+        try {
+            val parsedJson = JsonUtil.extractJsonObjectIntoMap(jsonPayload)
+
+            val correlationID = parsedJson["correlationID"]
+            Logger.info(methodTag, "Correlation ID during JavaScript Call: [$correlationID]")
+
+            // TODO: Leaving these here, as these will be relevant for next WebCP feature
+            // val actionName = parsedJson["action_name"]
+            // val actionComponent = parsedJson["action_component"]
+
+            val parameters = JsonUtil.extractJsonObjectIntoMap(parsedJson["params"])
+            val function = parameters["function"]
+            val data = JsonUtil.extractJsonObjectIntoMap(parameters["data"])
```

Comment:
Let's define a DTO representing the json scheme and then you can just use the kotlinx serialization libraries (or even GSON or Moshi if you prefer that) to deserialize the raw json directly to a custom dto.

For instance:

```java
@Serializable
data class AuthUxApiResponse(
    val correlationId: String,
    val actionName: String,
    val actionComponent: String,
    val params: Params; // define this as a DTO as well
)
```

and then you can deserialize as follows us kotlinx serialization framework

```java
val authUxResponse: AuthUxApiResponse = Json.decodeFromString(jsonString)
```

and then you can access individual properties as follows:

```java
   val numberMatch = authUxResponse.parameters.data.numberMatch;
```

Reply:
Neat! I can adjust to this, this current Json parsing implementation is a bit janky